src/dune/dune_file.ml

```diff
@@ -2150,6 +2148,30 @@ module Include_subdirs = struct
     enum opts_list
 end
 
+module Deprecated_library_name = struct
+  type t =
+    { loc : Loc.t
+    ; project : Dune_project.t
+    ; old_public_name : Public_lib.t
+    ; new_public_name : Lib_name.t
```

Comment:
Do you think it makes sense to try and share this type with `Dune_package` somehow? Even at the cost of a type parameter. I'm just thinking that we'd like to unify these types as much as possible now, so there's little appetite to introduce more duplication.

For this particular type, the difference seems to come from the extra `Dune_project.t`. I've thought about introduce the following stanza wrapper:

```ocaml
type stanza =
  { loc : Loc.t
  ; project : Dune_project.t
  ; stanza : Stanza.t
  }
```

and this would take care of location and project parsing for all stanzas in a uniform way. This is a common concern for most stanzas.

Reply:
This stanza type makes sense yes, I'm having a look at sharing the type with Dune_package

Reply:
Hmm, well there is the `Public_lib` thing. It might be possible to unify, but it looks like it's going to require some refactoring. It seems best left for another PR, what do you think?

Reply:
Yeah, that's fine. I can have a look at doing it as well.

Reply:
That'd be cool :)